Feature/update wc init UI

ahmed-tarek-salem opened the pull request:
**Changes**

- Fix the QR Scanner issue
- Update AddWalletConnect UI
- Add unsupported network validation
- Convert errors to appear in toasts

sallymoc commented:
Latest updates from Figma are missing as discussed in Discord

ahmed-tarek-salem commented:
@sallymoc 

Done, please check.

sallymoc commented:
@ahmed-tarek-salem please adjust the scanner view as indicated in Figma

sallymoc commented:
Also, as discussed in Discord, please add confirmation messages for the disconnect action:

Disconnect "<APP NAME>"?
Disconnecting this app will remove all access to your wallet

Disconnect all the apps?
Disconnecting the apps will remove all access to your wallet

Buttons: Cancel / Disconnect

sallymoc commented:
@ahmed-tarek-salem  The snackbar message pointed with the arrow is redundant and not needed (that screen already indicates there was a successfull scan). Please remove 
![image](https://github.com/user-attachments/assets/4d3f5e03-3a8f-4524-9668-7052f2728822)

sallymoc commented:
Please @ahmed-tarek-salem see also the Figma updates here

![image](https://github.com/user-attachments/assets/aa67f43c-4832-4dfc-bbea-aa4002db3f22)

sallymoc commented:
Also, @ahmed-tarek-salem  please ensure the WC functionality is always available (to remove any previous dependency with the enable/disable switch that was in settings in previous versions)

sallymoc commented:
Please also fix the layout for bigger screens here 
![image](https://github.com/user-attachments/assets/4c8ff102-8bfe-49c6-8528-c69a00eb4002)

Ahmed-Tarek-Designfy commented:
> @ahmed-tarek-salem The snackbar message pointed with the arrow is redundant and not needed (that screen already indicates there was a successfull scan). Please remove ![image](https://private-user-images.githubusercontent.com/231888/377061817-4d3f5e03-3a8f-4524-9668-7052f2728822.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNzUzNTgsIm5iZiI6MTcyOTE3NTA1OCwicGF0aCI6Ii8yMzE4ODgvMzc3MDYxODE3LTRkM2Y1ZTAzLTNhOGYtNDUyNC05NjY4LTcwNTJmMjcyODgyMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QxNDI0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NzYyNjgxZjdmMjU4NWRhNDJjMzA0MzE5MWE3MmRhMTFjOGMxYzQzYWNmZDVmMzUzZjI1ZDg0ZDZjMzkxYzZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.q4DxmEohw4Ltt7KpqiQZiClLeXFvZrTjjQi26B_LcO8)

There is a difference.

This snack bar appears immediately after detecting a proper QR code; it may or may not take some time to load before showing the new success screen.

ahmed-tarek-salem commented:
> @ahmed-tarek-salem The snackbar message pointed with the arrow is redundant and not needed (that screen already indicates there was a successfull scan). Please remove ![image](https://private-user-images.githubusercontent.com/231888/377061817-4d3f5e03-3a8f-4524-9668-7052f2728822.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMDAzNjAsIm5iZiI6MTcyOTIwMDA2MCwicGF0aCI6Ii8yMzE4ODgvMzc3MDYxODE3LTRkM2Y1ZTAzLTNhOGYtNDUyNC05NjY4LTcwNTJmMjcyODgyMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QyMTIxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NmVjNWI0ODJjMDMwNzA1YjE4ODY4ZTFmZTE0YTQxMWNjZmY2ODMzNjAzNmJiMzYxNThmMjUwZTA2MmMxYjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pSMkAl_5I7Urh4fL26P-Pldy73C4Ewxbzv7bC6q58mY)

There is a difference.

This snack bar appears immediately after detecting a proper QR code; it may or may not take some time to load before showing the new success screen.

If the loader (that appears on the Paste URL button) is enough to indicate to the user that a QR code has been detected, then we can safely remove that snack bar.

sallymoc commented:
> Please @ahmed-tarek-salem also fix the layout for bigger screens here ![image](https://private-user-images.githubusercontent.com/231888/377255627-4c8ff102-8bfe-49c6-8528-c69a00eb4002.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNjIxODgsIm5iZiI6MTcyOTI2MTg4OCwicGF0aCI6Ii8yMzE4ODgvMzc3MjU1NjI3LTRjOGZmMTAyLThiZmUtNDljNi04NTI4LWM2OWEwMGViNDAwMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxOFQxNDMxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZTk2MGU1YTcyZWQ1MGQ2MGVjMzRkNjQyOWIyYTVhNzM4NmEyMTVmYmExYzNhMTUyZWM0NzZhMGIwZTk1NmIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8ErS0Kc02Jg0ZK_8a4dwIJRHNupxHgEWuPtBjEkHGxo)

Resolved

sallymoc commented:
> @ahmed-tarek-salem Also, as discussed in Discord, please add confirmation messages for the disconnect action:
> 
> Disconnect ""? Disconnecting this app will remove all access to your wallet
> 
> Disconnect all the apps? Disconnecting the apps will remove all access to your wallet
> 
> Buttons: Cancel / Disconnect

Text for title and cancel button text doesn't match with the requirement in the current implementation

sallymoc commented:
> > @ahmed-tarek-salem The snackbar message pointed with the arrow is redundant and not needed (that screen already indicates there was a successfull scan). Please remove ![image](https://private-user-images.githubusercontent.com/231888/377061817-4d3f5e03-3a8f-4524-9668-7052f2728822.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMDAzNjAsIm5iZiI6MTcyOTIwMDA2MCwicGF0aCI6Ii8yMzE4ODgvMzc3MDYxODE3LTRkM2Y1ZTAzLTNhOGYtNDUyNC05NjY4LTcwNTJmMjcyODgyMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QyMTIxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NmVjNWI0ODJjMDMwNzA1YjE4ODY4ZTFmZTE0YTQxMWNjZmY2ODMzNjAzNmJiMzYxNThmMjUwZTA2MmMxYjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pSMkAl_5I7Urh4fL26P-Pldy73C4Ewxbzv7bC6q58mY)
> 
> There is a difference.
> 
> This snack bar appears immediately after detecting a proper QR code; it may or may not take some time to load before showing the new success screen.
> 
> If the loader (that appears on the Paste URL button) is enough to indicate to the user that a QR code has been detected, then we can safely remove that snack bar.

Yes, the loading indicator + the screen user's interaction disabled it's good enough for this @ahmed-tarek-salem

ahmed-tarek-salem commented:
> > @ahmed-tarek-salem Also, as discussed in Discord, please add confirmation messages for the disconnect action:
> > Disconnect ""? Disconnecting this app will remove all access to your wallet
> > Disconnect all the apps? Disconnecting the apps will remove all access to your wallet
> > Buttons: Cancel / Disconnect
> 
> Text for title and cancel button text doesn't match with the requirement in the current implementation

Resolved, removed unused IdListItem code, and added a shared ConfirmationDialog.

ahmed-tarek-salem commented:
> > > @ahmed-tarek-salem The snackbar message pointed with the arrow is redundant and not needed (that screen already indicates there was a successfull scan). Please remove ![image](https://private-user-images.githubusercontent.com/231888/377061817-4d3f5e03-3a8f-4524-9668-7052f2728822.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMDAzNjAsIm5iZiI6MTcyOTIwMDA2MCwicGF0aCI6Ii8yMzE4ODgvMzc3MDYxODE3LTRkM2Y1ZTAzLTNhOGYtNDUyNC05NjY4LTcwNTJmMjcyODgyMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxN1QyMTIxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NmVjNWI0ODJjMDMwNzA1YjE4ODY4ZTFmZTE0YTQxMWNjZmY2ODMzNjAzNmJiMzYxNThmMjUwZTA2MmMxYjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pSMkAl_5I7Urh4fL26P-Pldy73C4Ewxbzv7bC6q58mY)
> > 
> > 
> > There is a difference.
> > This snack bar appears immediately after detecting a proper QR code; it may or may not take some time to load before showing the new success screen.
> > If the loader (that appears on the Paste URL button) is enough to indicate to the user that a QR code has been detected, then we can safely remove that snack bar.
> 
> Yes, the loading indicator + the screen user's interaction disabled it's good enough for this @ahmed-tarek-salem

Resolved.

sallymoc commented:
Please resolve the conflicts

ahmed-tarek-salem commented:
@sallymoc 
Done
